runtime/compiler/net/ClientStream.cpp

```diff
@@ -130,60 +130,86 @@ SSL_CTX *ClientStream::_sslCtx = NULL;
 
 int openConnection(const std::string &address, uint32_t port, uint32_t timeoutMs)
    {
-   // TODO consider using newer API like getaddrinfo to support IPv6
-   struct hostent *entSer = gethostbyname(address.c_str());
-   if (!entSer)
+   // TODO consider support for IPv6
+   struct addrinfo hints;
+   memset(&hints, 0, sizeof(hints));
+   hints.ai_family = AF_INET;    // Allow IPv4 only; could use AF_UNSPEC to allow IPv6 too
+   hints.ai_socktype = SOCK_STREAM;
+   hints.ai_flags = 0;
+   hints.ai_protocol = 0; // Any protocol
+
+   char portName[12];
+   int r = snprintf(portName, 12, "%d", port);
+   if (r < 0 || r > 11)
+      throw StreamFailure("snprintf failed");
+
+   struct addrinfo *addrList = NULL;
+   int res = getaddrinfo(address.c_str(), portName, &hints, &addrList);
+   if (res != 0)
+      {
+      // Can use gai_strerror(res) to show error code
       throw StreamFailure("Cannot resolve server name");
-
-   struct sockaddr_in servAddr;
-   memset(&servAddr, 0, sizeof(servAddr));
-   memcpy(&servAddr.sin_addr.s_addr, entSer->h_addr, entSer->h_length);
-   servAddr.sin_family = AF_INET;
-   servAddr.sin_port = htons(port);
-
-   int sockfd = socket(AF_INET, SOCK_STREAM, 0);
+      }
+   struct addrinfo *pAddr;
+   int sockfd = -1;
+   for (pAddr = addrList; pAddr; pAddr = pAddr->ai_next) 
+      {
+      sockfd = socket(pAddr->ai_family, pAddr->ai_socktype, pAddr->ai_protocol);
+      if (sockfd >= 0)
+         break; // Use the first address for which a socket could be created
+      // Try next address
+      }
    if (sockfd < 0)
+      {
+      if (addrList)
```

Comment:
This check isn't done elsewhere and seems unnecessary and since `getaddrinfo` will return at least one element if it succeeded.

Reply:
I was thinking about the following sequence of events:
1. getaddrinfo returns an empty list (but does not return an error code)
2. The `for` loop is never executed, thus `sockfd==-1`
3. We enter this `if` statement and execute `freeaddrinfo(NULL)`. This may be benign, but I didn't want to take any chances.

The check `addrList != NULL` is not needed in other parts because we throw if a socket cannot be opened, and a socket being opened guarantees that `addrList != NULL`

Reply:
I removed the check